sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamAggregationRel.java

```diff
@@ -82,9 +82,44 @@ public BeamAggregationRel(
     this.windowFieldIndex = windowFieldIndex;
   }
 
+  private NodeStats computeWindowingCostEffect(NodeStats inputStat) {
+    if (windowFn == null) {
+      return inputStat;
+    }
+    WindowFn w = windowFn;
+    double multiplicationFactor = 1;
+    if (w instanceof SlidingWindows) {
+      multiplicationFactor =
+          ((double) ((SlidingWindows) w).getSize().getStandardSeconds())
+              / ((SlidingWindows) w).getPeriod().getStandardSeconds();
+    }
+
+    return NodeStats.create(
+        inputStat.getRowCount() * multiplicationFactor,
+        inputStat.getRate() * multiplicationFactor,
+        BeamIOSourceRel.CONSTANT_WINDOW_SIZE);
+  }
+
   @Override
   public NodeStats estimateNodeStats(RelMetadataQuery mq) {
```

Comment:
Can you add a comment describing the logic and the math happening here? E.g. it's unclear what's behind `groupCount` stuff

Reply:
Same in other rels